data/helpers.d/mysql

```diff
@@ -90,3 +90,47 @@ ynh_mysql_create_user() {
 ynh_mysql_drop_user() {
     ynh_mysql_execute_as_root "DROP USER '${1}'@'localhost';"
 }
+
+# Create a database, an user and its password. Then store the password in the app's config
+#
+# User of database will be store in db_user's variable.
+# Name of database will be store in db_name's variable.
+# And password in db_pwd's variable.
+#
+# usage: ynh_mysql_generate_db user name
+# | arg: user - Owner of the database
+# | arg: name - Name of the database
+ynh_mysql_generate_db () {
+	db_pwd=$(ynh_string_random)	# Generate a random password
+	ynh_mysql_create_db "$2" "$1" "$db_pwd"	# Create the database
+	ynh_app_setting_set $app mysqlpwd $db_pwd	# Store the password in the app's config
```

Comment:
Yes, db_pwd will not exist if you use this helper in a subshell. But, it's not the use case.
But, maybe you can return the password. It's more complicated, but, more evident. Really, as it is, It's ok for me.